optimism/src/keccak/tests.rs

```diff
@@ -470,3 +476,52 @@ fn test_keccak_multiplicities() {
         2
     );
 }
+
+// Prover/Verifier test includidng the Keccak constraints
+#[test]
+fn test_keccak_prover() {
+    // guaranteed to have at least 30MB of stack
+    stacker::grow(30 * 1024 * 1024, || {
+        let mut rng = o1_utils::tests::make_test_rng();
+        let domain_size = 1 << 8;
+
+        // Generate 3 blocks of preimage data
+        let bytelength = rng.gen_range(2 * RATE_IN_BYTES..RATE_IN_BYTES * 3);
+        let preimage: Vec<u8> = (0..bytelength).map(|_| rng.gen()).collect();
+
+        // Keep track of the constraints and lookups of the sub-circuits
+        let mut keccak_circuit = KeccakCircuit::<Fp>::new(domain_size);
+
+        // Initialize the environment and run the interpreter
+        let mut keccak_env = KeccakEnv::<Fp>::new(0, &preimage);
+
+        // We want to use domain_size rows, even if that is an incomplete Keccak execution
+        while keccak_env.constraints_env.step.is_some() {
+            let step = keccak_env.constraints_env.step.unwrap();
+            // Run the interpreter, which sets the witness columns
+            keccak_env.step();
+            // Check witness satisfies constraints
+            keccak_env.witness_env.constraints();
+
+            // Add the witness row to the circuit
+            keccak_circuit.push_row(step, &keccak_env.witness_env.witness.cols);
+        }
+        keccak_circuit.pad_rows();
+
+        for step in [
+            Round(0),
+            Sponge(Absorb(First)),
+            Sponge(Absorb(Middle)),
+            Sponge(Absorb(Last)),
+            Sponge(Absorb(Only)),
+            Sponge(Squeeze),
+        ] {
+            test_completeness_generic::<ZKVM_KECCAK_COLS, _>(
```

Comment:
So each step of Keccak is a separate "circuit" with its own set of constraints? 

If yes, does this circuit handle more than one round of Keccak? (probably no?)